cli/src/registry/edit-with-tambo-button/edit-with-tambo-button.tsx

```diff
@@ -0,0 +1,407 @@
+export function EditWithTamboButton({
+  icon,
+  tooltip = "Edit with tambo",
+  description,
+  className,
+  onOpenThread,
+  editorRef,
+}: EditWithTamboButtonProps) {
+  const component = useTamboCurrentComponent();
+  const { sendThreadMessage, isIdle } = useTambo();
+
+  const [prompt, setPrompt] = useState("");
+  const [isPending, setIsPending] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [sendMode, setSendMode] = useState<"send" | "thread">("send");
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const wasGeneratingRef = useRef(false);
+
+  // If no component, the current component is not an interactable - don't render.
+  if (!component) {
+    return null;
+  }
+
+  const isGenerating = !isIdle;
+
+  // Close popover when generation completes
+  useEffect(() => {
+    if (isGenerating) {
+      wasGeneratingRef.current = true;
+    } else if (wasGeneratingRef.current && !isPending) {
+      // Generation just completed - close the popover
+      wasGeneratingRef.current = false;
+      setIsOpen(false);
+      setPrompt("");
+    }
+  }, [isGenerating, isPending]);
+
+  const handleSend = useCallback(async () => {
+    if (!prompt.trim() || isPending) {
+      return;
+    }
+
+    setIsPending(true);
+    setError(null);
+
+    try {
+      // Send the message with the interactable component in context
+      await sendThreadMessage(prompt.trim(), {
+        streamResponse: true,
+        additionalContext: {
+          inlineEdit: {
+            componentId: component.interactableId,
+            instruction:
+              "The user wants to edit this specific component inline. Please update the component's props to fulfill the user's request.",
+          },
+        },
+      });
+
+      // Clear the prompt after successful send
+      setPrompt("");
+    } catch (err) {
+      const error =
+        err instanceof Error ? err : new Error("Failed to send message");
+      setError(error);
+    } finally {
+      setIsPending(false);
+    }
+  }, [prompt, isPending, component, sendThreadMessage]);
+
+  const handleSendInThread = useCallback(() => {
+    if (!prompt.trim()) {
+      return;
+    }
+
+    // Save the message before clearing
+    const messageToInsert = prompt.trim();
+
+    // NOTE: This registry version uses simple text insertion for portability.
+    // It does NOT use TipTap Mention nodes or context attachments to avoid
+    // dependencies on specific editor configurations. Users can implement
+    // their own context system if needed.
+
+    // Open the thread panel if callback provided
+    if (onOpenThread) {
+      onOpenThread();
+    }
+
+    // Clear the prompt and close the modal
+    setPrompt("");
+    setIsOpen(false);
+
+    // Insert text into the editor if editorRef is provided
+    if (editorRef?.current) {
+      setTimeout(() => {
```

Comment:
hmm not a fan of the timeout here. I wonder if we can preload the editor and have the editor autofocus?

I think any time we're reaching from a parent -> child to invoke something, that's a codesmell.